File: 2021-01-01/sol.py
import re

nthEven = lambda n:2*n-2

print(nthEven(1))
print(nthEven(2))
print(nthEven(100))

validateEmail = lambda s:bool(re.match(r".+@.+\..+", s))

print(validateEmail("@gmail.com"))
print(validateEmail("hello.gmail@com"))
print(validateEmail("gmail"))
print(validateEmail("hello@gmail"))
print(validateEmail("hello@edabit.com"))

atbash=lambda  s:''.join([chr(90-(ord(x)-65)) if x.isalpha() and x.isupper() else (chr(122-(ord(x)-97) if x.isalpha() else ord(x))) for x in s])

print(atbash("apple"))
print(atbash("Hello world!"))
print(atbash("Christmas is the 25th of December"))

File: 2021-01-01/sol.cpp
#include <iostream>
#include <string>
#include <regex>

int nthEven(int n) {
    return 2*n-2;
}

bool validateEmail(std::string s) {
    return regex_match(s,std::regex("^.+@.+\\..+$"));
}

std::string atbash(std::string s) {
    std::string r;
    for(char x:s) {
        if(isalpha(x))
            if(islower(x))
                r+=219-x;
            else
                r+=155-x;
        else
            r+=x;
    }
    return r;
}

int main() {
    std::cout << nthEven(1) << std::endl;
    std::cout << nthEven(2) << std::endl;
    std::cout << nthEven(100) << std::endl;
    std::cout << std::boolalpha << validateEmail("@gmail.com") << std::endl;
    std::cout << std::boolalpha << validateEmail("hello.gmail@com") << std::endl;
    std::cout << std::boolalpha << validateEmail("gmail") << std::endl;
    std::cout << std::boolalpha << validateEmail("hello@gmail") << std::endl;
    std::cout << std::boolalpha << validateEmail("hello@edabit.com") << std::endl;
    std::cout << atbash("apple") << std::endl;
    std::cout << atbash("Hello world!") << std::endl;
    std::cout << atbash("Christmas is the 25th of December") << std::endl;
    return 0;
}


